### DISCUSSION TITLE
SwiftUINagivation SPM + TCA please unexpected behavior 🥲

### DISCUSSION THREAD

**Discussion by saroar** [pointfreeco/swift-composable-architecture#965] [Q&A]
I am trying to use SwiftUINavigation with SPM but I am getting huge bug if anyone will have time please check it also any refactor advice will be appreciated 

screen video + full source code include here

**Comment by saroar** [comment]
[MediaparkNewsSwiftUINavBarBug.zip](https://github.com/pointfreeco/swift-composable-architecture/files/7864511/MediaparkNewsSwiftUINavBarBug.zip)


https://user-images.githubusercontent.com/8770772/149375758-69037115-8fd8-418a-a240-9b45559e1f8e.mp4

**Comment by saroar** [comment]


**Reply by iampatbrown** [reply]
@saroar Just from a quick look at your updated code. You probably don't need `ViewState` and `ViewAction`. Instead use

```swift
public var body: some View {
  WithViewStore(store.stateless) { viewStore in
    ZStack { 
    ...
```

The next thing you will need to do is ensure parent views are not observing changes to `SearchBarFilterSectorsState`. For example, right now `SearchBarFilterView` will update when changes are made to `SearchBarFilterSectorsState`. This is at least one reason you are seeing the current behaviour.

**Reply by saroar** [reply]
so I need to add in my parent view ? need ViewState and ViewAction fit?

**Reply by saroar** [reply]
@iampatbrown if you download my app should i update my DatePickerWithButtons View ? with TCA?

**Reply by saroar** [reply]
here is update version still same issue 
[MediaparkNewsNavBarUnexpectedBehaive.zip](https://github.com/pointfreeco/swift-composable-architecture/files/7871385/MediaparkNewsNavBarUnexpectedBehaive.zip)
